File: Problems 2100-2199/2140. Solving Questions With Brainpower/solution.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& questions) {
        int n = questions.size();
        vector<long long> dp(n+1, 0);
        for (int i = n-1; i >= 0; i--) {
            long long solve = questions[i][0];
            if (i+questions[i][1]+1 < n) {
                solve += dp[i+questions[i][1]+1];
            }
            dp[i] = max(solve, dp[i+1]);
        }
        return dp[0];
    }
};

File: Problems 2100-2199/2140. Solving Questions With Brainpower/solution.py
class Solution:
    def mostPoints(self, questions: List[List[int]]) -> int:
        n = len(questions)
        dp = [0]*(n+1)

        for i in range(n-1, -1, -1):
            solve = questions[i][0]
            if i + questions[i][1] + 1 < n:
                solve += dp[i+questions[i][1]+1]
            dp[i] = max(solve, dp[i+1])
        return dp[0]